src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+use Doctrine\ORM\ORMException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\BlogPostRepository;
+
+class BlogPostController extends AbstractController
+{
+
+    /**
+     * @Route("/blogposts/{userID?}", name="blog-posts", methods={"GET"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     * returns blogposts by userID or If userID null, all blogposts
+     */
+    public function getBlogPostList(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
+        ?int $userID = null,
+    ):
+    Response {
+        /** @var BlogPostListEnquiry $enquiry */
+        $enquiry = $serializer->deserialize(
+            $request->getContent(),
+            BlogPostListEnquiry::class,
+            'json'
+        );
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+        $blogPostService = new BlogPostService($blogPostRepository);
+
+        $posts = $blogPostService->getBlogPosts($userID, $enquiry->getLimit(), $enquiry->getOffset());
+
+        $posts_json = json_encode(
+            array_map(array($blogPostService, 'getBlogPostReturnData'), $posts),
+            JSON_THROW_ON_ERROR
+        );
+
+        return new Response($posts_json, 200, ['Content-Type' => 'application/json']);
+    }
+
+    /**
+     * @Route("/blogposts/{userID}/create", name="blog-posts-create", methods={"POST"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     */
+    public function createBlogPost(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
+        int $userID,
+    ):
+    Response {
+        /** @var BlogPostEnquiry $postEnquiry */
+        $postEnquiry = $serializer->deserialize(
+            $request->getContent(),
+            BlogPostEnquiry::class,
+            'json'
+        );
+
+        /** @var BlogUserRepository $blogPostRepository */
+        $blogUserRepository = $entityManager->getRepository(BlogUser::class);
+        $user = $blogUserRepository->find($userID);
+
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+        $blogPostService = new BlogPostService($blogPostRepository);
+
+        if (!$user) {
+            throw $this->createNotFoundException(
+                'No user found for id' . $userID
+            );
+        }
+
+        if (empty($postEnquiry->getBody())) {
+            return new Response(
+                json_encode(["success" => false, "msg" => "Missing body text"], JSON_THROW_ON_ERROR),
+                400,
+                ['Content-Type' => 'application/json']
+            );
+        }
```

Comment:
[Jeff] check framework validation for symfony, laravel has more concise way of checking/validating input, symfony might have one too

Reply:
There's a [validate service](https://symfony.com/doc/4.4/validation.html#installation) to assert conditions and validating them by calling `validate()`